titan-server/src/bin/main.rs

```diff
@@ -1,16 +1,32 @@
-#![feature(proc_macro_hygiene, decl_macro)]
+#![feature(plugin, decl_macro, custom_attribute, proc_macro_hygiene)]
 
 extern crate libtitan;
 #[macro_use] extern crate rocket;
-extern crate rocket_contrib;
+use rocket::fairing::AdHoc;
+#[macro_use] extern crate rocket_contrib;
+extern crate regex;
+extern crate bcrypt;
+extern crate frank_jwt;
+extern crate rocket_cors;
 
 use libtitan::routes;
 use libtitan::db;
+use libtitan::config;
+use libtitan::cors_fairing;
 
 fn main() {
     rocket::ignite()
         .attach(db::TitanPrimary::fairing())
         .attach(db::UnksoMainForums::fairing())
+        .attach(cors_fairing::cors())
+        .attach(AdHoc::on_attach("Auth", |rocket| {
+            let settings = config::AuthConfig::from_env(&rocket.config());
+            match settings {
```

Comment:
This `match` exists `from_env` returns a `Result<AuthConfig, ConfigError>`, which was being stored in state as-is. This was what was causing the issue at run-time in the auth handler. We needed to unwrap the result and return an `Ok(rocket.manage(...))` or `Err(rocket)` based on the success of fetching the auth key from the config.

Rather than call `unwrap`, which will cause a panic at runtime, I decided to use a `match` to return a `Result<_, _>` that Rocket can use to report whether attaching the fairing was successful or not.